lib/kvdb/kvdb_rest.c

```diff
@@ -871,38 +871,54 @@ kvdb_rest_add_endpoints(struct ikvdb *const kvdb)
     alias = ikvdb_alias(kvdb);
 
     status = rest_server_add_endpoint(0, handlers[0], kvdb, "/kvdbs/%s/compact", alias);
-    if (ev(status) && !err)
+    if (ev(status) && !err) {
```

Comment:
If this fails, we bring down all of HSE. Presumably, this is not what we want. Do others agree?

Reply:
It's called during `kvdb_open`, and most of the failure cases seem to be developer error (sprintf overflow, duplicated enpoints, etc).  I would say failing the `kvdb_open` call is the correct action.

Reply:
If you look in hse_init(), you can see I am ignoring errors. Should I stop doing that then?

Reply:
As currently coded, these failures will not cause opens to fail.  You'd have to change `ikvdb_rest_register` to pass the error to `ikvdb_open`.

Reply:
Nice catch...I will submit a separate PR to fail things then.